### ISSUE TITLE
Upgrade for Umbraco 9

### ISSUE THREAD

**Issue by creativesuspects** [mjbarlow/Font-Awesome-Links-Property-Editor#10]
@mjbarlow This package could easily be upgrade to Umbraco 9 with just a few minor changes. I haven't had time to create a pull request (it was quicker to just include the code for the Models and PropertyValueConverters) in my current project. But you can download the updated code for the Models and PropertyValueConverters here:

https://www.dropbox.com/s/d4s1pjlegjvygvx/FaLinksPropertyEditor_v9.zip?dl=0

**Comment by lundmorten** [comment]
Hi @creativesuspects, can you please elaborate how to update, using your code?

**Comment by creativesuspects** [comment]
Hi @lundmorten,

Just copy the App_Plugins/FaLinksPropertyEditor/ folder from the Github source into your Umbraco v9 project:

https://github.com/mjbarlow/Font-Awesome-Links-Property-Editor/tree/master/FaLinksPropertyEditor/App_Plugins/FaLinksPropertyEditor

And include the Models and PropertyValueConverters from my .zip package in the project and build the solution. You should then be able to create/use the available data types in this package:

Font Awesome Icon
Font Awesome Icon Buttons
Font Awesome Links

I may have some spare time to create a PR later this week.
Hopefully @mjbarlow is able to release version 2.x for Umbraco 9+ using my PR.

**Comment by jamiehowarth0** [comment]
Am currently working on bringing this from a U8 project to U10 - I may take a look at your code @creativesuspects to try and produce an intermediary V9 package if I can.

**Comment by creativesuspects** [comment]
@jamiehowarth0 I’ve actually created a custom package based on FontAwesomeLinksPropertyEditor with a bit more flexibility: it’s very easy to define multiple data types using different SVG sprites for the icon source (instead of a single json file). Also, when using SVG sprites you don’t have to include any webfonts for the icons. This custom package is compatible with Umbraco 10 and 11. I’d be happy to share the source code with you if you want.

I’ve only published this package on a private nuget server, but feel free to use my code to create your own nuget package.

**Comment by jamiehowarth0** [comment]
@creativesuspects yes please! Better still - are you able to publish it as a Github repo? Happy to help turn it into a Nuget package from that - I'm trying to avoid having to compile third-party packages from source for my client project.

**Comment by creativesuspects** [comment]
@jamiehowarth0 Sure, I’ll do some project cleanup tomorrow and I’ll publish it to Github. I’ll let you know when the repo is available.

**Comment by jamiehowarth0** [comment]
@creativesuspects thankyou so much! I look forward to seeing it 😁

**Comment by creativesuspects** [comment]
@jamiehowarth0 Here you go: https://github.com/creativesuspects/SvgIconPropertyEditor

**Comment by mjbarlow** [comment]
Thanks, am archiving this repo. As am no longer supporting it. Looks like @creativesuspects has a great alternative!
